lib/wasix/src/journal/effector/syscalls/path_rename.rs

```diff
@@ -26,16 +28,26 @@ impl JournalEffector {
         new_fd: Fd,
         new_path: &str,
     ) -> anyhow::Result<()> {
-        let ret = crate::syscalls::path_rename_internal(ctx, old_fd, old_path, new_fd, new_path)?;
-        if ret != Errno::Success {
-            bail!(
-                "journal restore error: failed to rename path (old_fd={}, old_path={}, new_fd={}, new_path={}) - {}",
-                old_fd,
-                old_path,
-                new_fd,
-                new_path,
-                ret
-            );
+        if old_fd == VIRTUAL_ROOT_FD && new_fd == VIRTUAL_ROOT_FD {
```

Comment:
As above, why is this needed, needs an explanatory comment. 
Can we avoid the custom logic?

Reply:
See above